### ISSUE TITLE
WebSockets connection won't close.

### ISSUE THREAD

**Issue by michielboekhoff** [mirek/CoreWebSocket#12]
Using https://www.websocket.org/echo.html and running the example on my MacBook Pro, the connection will refuse to close. 

AppDelegate.m:

```
#import "AppDelegate.h"
#include "CoreWebSocket/CoreWebSocket.h"

@interface AppDelegate ()

@property (weak) IBOutlet NSWindow *window;
@end

@implementation AppDelegate

 void Callback(WebSocketRef self, WebSocketClientRef client, CFStringRef value) {
    if (value) {
        CFShow(value);
        WebSocketWriteWithString(self, CFSTR("hello!"));
    }
}

- (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
    // Insert code here to initialize your application
    WebSocketRef webSocket = WebSocketCreateWithHostAndPort(NULL, kWebSocketHostAny,     6001, NULL);
    if (webSocket) {
        webSocket->callbacks.didClientReadCallback = Callback;
    }
}

- (void)applicationWillTerminate:(NSNotification *)aNotification {
    // Insert code here to tear down your application
}

@end
```

I tried, and the callback willRemoveClientCallback is not called. Any help with this?

Alex.

**Comment by michielboekhoff** [comment]
I see pull request #10 fixes part of this problem! Is anyone going to merge that? Right now, the only thing that needs to be done is sending a close frame.

**Comment by michielboekhoff** [comment]
I forked the repo and merged the changes with #10 (more or less).

**Comment by michielboekhoff** [comment]
Closed as this is _very_ old.
